tests/kafkatest/services/kafka/kafka.py

```diff
@@ -165,6 +165,24 @@ def start(self, add_principals=""):
         self.start_minikdc(add_principals)
         Service.start(self)
 
+        self.logger.info("Waiting for brokers to register at ZK")
+
+        retries = 30
+        expected_broker_ids = set(self.nodes)
+        broker_ids = set()
+        while retries > 0:
```

Comment:
normally in the tests we use `wait_until` instead of manually doing retries, and it generally makes the entire thing a one-liner. it's also generally better since you can just set time-based termination conditions instead of coupling # of retries with the timeout.